### ISSUE TITLE
morebits: add live preview 

### ISSUE THREAD

**Issue by siddharthvp** [azatoth/twinkle#1050]
It would be good to have the infrastructure for live previews in morebits. I'd imagine this would be useful primarily for previewing comments (edit/log summaries) where we usually offer only a text field rather than a textarea, in which it's difficult to correctly write something long with links etc. Mentioned by @roysmith on  [WT:TW](https://en.wikipedia.org/wiki/Wikipedia_talk:Twinkle/Archive_43#U/I_problems_for_CSD_with_multiple_reasons.).

Relevant modules which use summaries that ought to be previewed: speedy, unlink, batchdelete, batchundelete, batchprotect. 

It doesn't look as if we can cleanly implement this as an additional option in `Morebits.wiki.preview`. So what I would do is to create a new `Morebits.livePreview` (or `Morebits.wiki.livePreview` if you're a pedant).

**Comment by Amorymeltzer** [comment]
Sorry, I'm not entirely following; I think you're suggesting a live-updating preview of edit (or log) summary fields.  Is that right?

**Comment by roysmith** [comment]
Yes. the coolest thing would be a live-updating preview.  But, even just having a "preview" button that you had to click to get a preview would a big improvement, and from my point of view, good enough.

**Comment by siddharthvp** [comment]
> Sorry, I'm not entirely following; I think you're suggesting a live-updating preview of edit (or log) summary fields. Is that right?

Yes. I think live-updating previews are better for summary fields to avoid the UI clutter of a preview button. Do you think it would be too jarring? Notably, I think the new Reply tool uses live preview.
